lib/archivist/encoders.js

```diff
@@ -21,7 +32,13 @@ exports.registerCommonEncoder = function (fromEncoding, toEncoding, fn) {
 
 	var name = fromEncoding + '-' + toEncoding;
 
-	commonEncoders[name] = fn;
+	commonEncoders[name] = function (data) {
```

Comment:
Can't data be an actual object if the `fromEncoding` is `live` ?

Reply:
From what I can tell, `registerCommonEncoder` is exposed but only called internally (somehow). And internally, all encoding rules are from string formats to buffers or the opposite. @ronkorving might be able to confirm my understanding of this bit of code.

Reply:
Yeah.. I think it would be more appropriate to do this for each individual encoder function. That would also imply they will only have to test for *one* type, not both strings and buffers. The code below implies I can still pass the wrong type to an encoder (if it expects a buffer, I can send it a string).

Reply:
From what I can tell, all encoders can take both string or Buffer:

```shell
> new Buffer('test')
<Buffer 74 65 73 74>
> new Buffer(new Buffer('test'))
<Buffer 74 65 73 74>
> 'asdas'.toString()
'asdas'
>
```

Reply:
Ah, spoke too soon. A string `toString` is not working the same way as the `Buffer` object method.